python/pylibcugraph/pylibcugraph/k_truss_subgraph.pyx

```diff
@@ -0,0 +1,163 @@
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.graph_functions cimport (
+    cugraph_induced_subgraph_result_t,
+    cugraph_induced_subgraph_get_sources,
+    cugraph_induced_subgraph_get_destinations,
+    cugraph_induced_subgraph_get_edge_weights,
+    cugraph_induced_subgraph_get_subgraph_offsets,
+    cugraph_induced_subgraph_result_free,
+)
+from pylibcugraph._cugraph_c.community_algorithms cimport (
+    cugraph_k_truss_subgraph,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+)
+
+
+def k_truss_subgraph(ResourceHandle resource_handle,
+                     _GPUGraph graph,
+                     size_t k,
+                     bool_t do_expensive_check):
+    """
+    Extract k truss of a graph for a specific k.
+
+    Parameters
+    ----------
+    resource_handle : ResourceHandle
+        Handle to the underlying device resources needed for referencing data
+        and running algorithms.
+
+    graph : SGGraph
+        The input graph.
+
+    k: size_t
+        The desired k to be used for extracting the k-truss subgraph.
+
+    do_expensive_check : bool_t
+        If True, performs more extensive tests on the inputs to ensure
+        validitity, at the expense of increased run time.
+
+    Returns
+    -------
+    A tuple of device arrays containing the sources, destinations,
+    edge_weights and edge_offsets.
+
+    Examples
+    --------
+    >>> import pylibcugraph, cupy, numpy
+    >>> srcs = cupy.asarray([0, 1, 1, 3, 1, 4, 2, 0, 2, 1, 2,
+    ...     3, 3, 4, 3, 5, 4, 5], dtype=numpy.int32)
+    >>> dsts = cupy.asarray([1, 0, 3, 1, 4, 1, 0, 2, 1, 2, 3,
+    ...     2, 4, 3, 5, 3, 5, 4], dtype=numpy.int32)
+    >>> weights = cupy.asarray(
+    ...     [0.1, 0.1, 2.1, 2.1, 1.1, 1.1, 7.2, 7.2, 2.1, 2.1,
+    ...     1.1, 1.1, 7.2, 7.2, 3.2, 3.2, 6.1, 6.1]
+    ...     ,dtype=numpy.float32)   
+    >>> k = 2
+    >>> resource_handle = pylibcugraph.ResourceHandle()
+    >>> graph_props = pylibcugraph.GraphProperties(
+    ...     is_symmetric=True, is_multigraph=False)
+    >>> G = pylibcugraph.SGGraph(
+    ...     resource_handle, graph_props, srcs, dsts, weights,
+    ...     store_transposed=False, renumber=False, do_expensive_check=False)
+    >>> (sources, destinations, edge_weights, subgraph_offsets) =
+    ...     pylibcugraph.k_truss_subgraph(resource_handle, G, k, False)
```

Comment:
I get an error when I execute this line
>>> (sources, destinations, edge_weights, subgraph_offsets) =  pylibcugraph.k_truss_subgraph(resource_handle, G, k, False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pylibcugraph' has no attribute 'k_truss_subgraph'